### PR TITLE
Table Row Details

### LINKED ISSUE #250: [Feature] GoTable Detail Row
### What is the feature that you would like to propose?
The ability to provide further details about a row of data in the table through a clickable arrow on the left most section of the row that expands a container element below said row. This container would be custom templated per the implementer.

_**Example:**_
The Kendo framework has a great working example of this:
https://www.telerik.com/kendo-angular-ui/components/grid/advanced-features/detail-template/

#### This feature will needs to have the following requirements:
- [x] Ability to render a row of detailed data associated with a parent row
- [x] It should be collapsable via a clickable arrow in the left most column of the parent rows
  - [x] By default, it will be collapsed
  - [x] By default, the arrow will point right
  - [x] When the row is expanded, the arrow should point down
- [ ] When the arrow is clicked, the table should emit a custom event to include:
  - [ ] The data of the row
  - [ ] Which direction the action took, expand/collapse
- [x] The detail row should be declared and formatted when building the table HTML (similar to the ability to format table cells and headers)
##### Extras:
- Provide animations for expanding/collapsing
- Provide a loader overlay for the detail row (used when making ajax requests after row is expanded)

### PR DESCRIPTION
**Pull request by StevenUlmer**
This adds the ability to use details within a table row. The implementor can provide a template
which is then used on every row to show the details when toggled on.

Fixes #250

### CODE REVIEW

File: projects/go-lib/src/lib/animations/table-details.animation.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export const detailsAnimation: AnimationTriggerMetadata = trigger('detailsAnimation', [
+  state('open', style({
+    display: 'table-row'
+  })),
+  state('close', style({
+    display: 'none'
+  })),
+  transition('open <=> close', [
+    animate('.125s ' + easing)
```
Comment (grahamhency) [inline projects/go-lib/src/lib/animations/table-details.animation.ts]: Do you feel like we need to have a different timing for this?

File: projects/go-lib/src/lib/components/go-table/go-table.component.html [hunk 2]
```diff
@@ -33,27 +37,46 @@ <h5 class="go-heading-5 go-heading--no-margin" *ngIf="tableTitle">{{ tableTitle
...
-            </ng-template>
-          </td>
-        </tr>
+        <ng-container *ngFor="let item of setDisplayData()">
+          <tr class="go-table__body-row">
+            <td class="go-table__body-column"
+                [ngClass]="{ 'go-table__body-row--expanded': item.showDetails }"
+                width="40"
+                *ngIf="details">
+              <go-icon-button
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-table/go-table.component.html]: So looking at this in the UI, the padding on the `icon-button` is making the row heights much taller than they probably should be. Do you think you could try to find a way around this?

File: projects/go-lib/src/lib/animations/table-details.animation.ts (hunk 1 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/animations/table-details.animation.ts]: Also, you can't perform an animation on `display` so this doesn't really do much visually.

File: projects/go-lib/src/lib/components/go-table/go-table.component.scss [hunk 3]
```diff
@@ -213,3 +213,11 @@
     text-align: center;
   }
 }
+
+.go-table__body-row--expanded {
+  border-bottom: none;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-table/go-table.component.scss]: What do you think about changing this to change the opacity of the color of the bottom border? That way we can animate the color and basically fade out the border.

File: projects/go-lib/src/lib/components/go-table/go-table.component.html (hunk 2 above)
Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-table/go-table.component.html]: I think I fixed it by adding a class that offsets the padding of the row by the amount that's on the `icon-button`

**Pull request merged**